libraries/ESP8266WebServer/src/Parsing-impl.h

```diff
@@ -62,215 +62,182 @@ static bool readBytesWithTimeout(typename ServerType::ClientType& client, size_t
 #endif
-    return false;
-  }
+        return false;
+    }
 
-  String methodStr = req.substring(0, addr_start);
-  String url = req.substring(addr_start + 1, addr_end);
-  String versionEnd = req.substring(addr_end + 8);
-  _currentVersion = atoi(versionEnd.c_str());
-  String searchStr = "";
-  int hasSearch = url.indexOf('?');
-  if (hasSearch != -1){
-    searchStr = url.substring(hasSearch + 1);
-    url = url.substring(0, hasSearch);
-  }
-  _currentUri = url;
-  _chunked = false;
-
-  HTTPMethod method = HTTP_GET;
-  if (methodStr == F("HEAD")) {
-    method = HTTP_HEAD;
-  } else if (methodStr == F("POST")) {
-    method = HTTP_POST;
-  } else if (methodStr == F("DELETE")) {
-    method = HTTP_DELETE;
-  } else if (methodStr == F("OPTIONS")) {
-    method = HTTP_OPTIONS;
-  } else if (methodStr == F("PUT")) {
-    method = HTTP_PUT;
-  } else if (methodStr == F("PATCH")) {
-    method = HTTP_PATCH;
-  }
-  _currentMethod = method;
+    String methodStr = req.substring(0, addr_start);
+    String url = req.substring(addr_start + 1, addr_end);
+    _currentVersion = req.substring(addr_end + 8).toInt();
+    String searchStr = "";
+    int hasSearch = url.indexOf('?');
+    if (hasSearch != -1) {
+        searchStr = url.substring(hasSearch + 1);
+        url = url.substring(0, hasSearch);
+    }
+    _currentUri = url;
+    _chunked = false;
+
+    HTTPMethod method = HTTP_GET;
+    if (methodStr == F("HEAD")) {
+        method = HTTP_HEAD;
+    } else if (methodStr == F("POST")) {
+        method = HTTP_POST;
+    } else if (methodStr == F("DELETE")) {
+        method = HTTP_DELETE;
+    } else if (methodStr == F("OPTIONS")) {
+        method = HTTP_OPTIONS;
+    } else if (methodStr == F("PUT")) {
+        method = HTTP_PUT;
+    } else if (methodStr == F("PATCH")) {
+        method = HTTP_PATCH;
+    }
+    _currentMethod = method;
 
 #ifdef DEBUG_ESP_HTTP_SERVER
-  DEBUG_OUTPUT.print("method: ");
-  DEBUG_OUTPUT.print(methodStr);
-  DEBUG_OUTPUT.print(" url: ");
-  DEBUG_OUTPUT.print(url);
-  DEBUG_OUTPUT.print(" search: ");
-  DEBUG_OUTPUT.println(searchStr);
+    DEBUG_OUTPUT.print("method: ");
+    DEBUG_OUTPUT.print(methodStr);
+    DEBUG_OUTPUT.print(" url: ");
+    DEBUG_OUTPUT.print(url);
+    DEBUG_OUTPUT.print(" search: ");
+    DEBUG_OUTPUT.println(searchStr);
 #endif
 
-  //attach handler
-  RequestHandlerType* handler;
-  for (handler = _firstHandler; handler; handler = handler->next()) {
-    if (handler->canHandle(_currentMethod, _currentUri))
-      break;
-  }
-  _currentHandler = handler;
+    //attach handler
+    RequestHandlerType* handler;
+    for (handler = _firstHandler; handler; handler = handler->next()) {
+        if (handler->canHandle(_currentMethod, _currentUri))
+            break;
+    }
+    _currentHandler = handler;
 
-  String formData;
-  // below is needed only when POST type request
-  if (method == HTTP_POST || method == HTTP_PUT || method == HTTP_PATCH || method == HTTP_DELETE){
```

Comment:
Did you test that moving this condition still results in correct function as before for all methods?

Reply:
I did not test all methods. I did test GET and POST though in the scope of my application and it worked fine. 

Another way to put it is however that this "if condition" didn't actually move. what happened is that the header parsing part that was used in the POST/PUT/PATCH/DELETE code was moved up front, and then the rest of the code remained the same. this allowed to remove the header parsing that was done for GET/HEAD case as it was duplicate now. 

Personally I think thats a relatively low risk change because the two header parsing loops were doing mostly the same, except that POST/PUT/PATCH/DELETE was checking for additional headers and parsed them. It now does so in all cases, but since it is extremely unlikely to get the additional header fields send by GET/HEAD requests, it will not matter in practice. and even if we get them, the parsed result is ignored as the if() condition remained the same and the parsed resulting values are not further evaluated.